driver/sdk.md

```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
 
 ```bash
 $ bblfsh-sdk prepare-build
 $ make build
 ```
 
-TODO
+You'll see a sequence of `docker build` and `docker run` commands execute and any
+building error will show on the standard output.
+
+### Configuring the run Dockerfile
+
+The second dockerfile template that you have to edit is the one that the Babelfish
+server will use to run new instances of drivers. Like the build one it will also 
+be generated by the `bblfsh-sdk` command but with the filename `Dockerfile.tpl`
+and this initial content without comments:
+
+```Dockerfile
+FROM alpine:3.5
+CMD /opt/driver/bin/driver
+```
+
+As with the build container, the first step is to add any packages needed to
+**run** the `native` binary produced by the build process, if any is needed like
+dynamically loading libraries for compiled languages (not included in Alpine base)
+or runtimes for interpreted or compiled-to-bytecode languages.
+
+Then you will need to copy the `build` directory into `/opt/driver/bin`, so that
+both the (automatically build by the SDK) `driver` and `native` binaries will 
+rest in that directory.
 
-## Testing
+The final `CMD` command should remain at the end of the file and you should not
+change it.
+
+### Creating the integration tests
+
+The integration tests will test all the process of the driver from the request
+to the parser to the annotated UAST generation. They work by comparing the 
+UAST output with the previously existing one, and will fail if any difference 
+is found.
+
+For this to work, the driver developer will need to provide files with source
+code examples in the `tests/` directory. with the `.source` file extension.
+Once this is done, you need to run the integration-test Makefile's target for the
+first time to automatically generate the `.native` and `.uast` files:
 
 ```bash
-$ bblfsh-sdk prepare-build
 $ make integration-test
 ```
 
-TODO
+The first time (or every time you add a new `.source` file) you'll need to
+manually check carefully the `.uast` files because they'll be used as reference on
+next runs. Once you're happy with them, any subsequent executions of the `make
+integration-test` will actually compare them.
+
+It's advisable to create very small `.source` files just testing the annotation of
+one language feature, or even several if the feature is complex. This will make
+the tests more atomic and help immensely with your sanity when trying to debug
+failed integration tests.
+
+### Officially adding your driver to Babelfish
+
+The last step if to have your language added to the [Babelfish Github's
+Project](https://github.com/bblfsh) so more people can use/break/debug or improve
+it.  For this you need to request a new repository to be created which you can
+make by opening a PR or talking with us in the [public Babelfish Slack
+channel](https://sourced-community.slack.com/join/shared_invite/MTkwNTM0ODEyODIzLTE0OTYxMzc5NTMtODRhMDYyNzAyYQ).
+Once the driver subproject has been added you'll make a PR with your code.
```

Comment:
I think we should define what an author should expect from babelfish before asking them to follow this process.

> Usually making a PR to an open source project doesn't make you automatically part of the organization or admin of the project so I don't think we need to explain that.

That's correct. It usually doesn't involve asking a project to create a new repo for your project, either, so it's a more involving process that just *making a PR*, it involves project ownership transfer.